### PR TITLE
Scrollbar migration guide

### PR DESCRIPTION
**Pull request by Piinks**
This change adds a migration guide for using the new ScrollBehaviors in order to control automatic Scrollbars.

### CODE REVIEW

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+---
+title: Default Scrollbars on Desktop
+description: ScrollBehaviors will now automatically build Scrollbars on Desktop platforms.
+---
+
+## Summary
+
+`ScrollBehavior`s will now automatically apply `Scrollbar`s to
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:8]: Delete "will".

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+scrolling widgets on desktop platforms - Mac, Windows and Linux.
+
+## Context
+
+Prior to this change, `Scrollbar`s were applied to scrolling widgets
+manually by the developer across all platforms. This did not match
+developer expectations when executing Flutter applications on desktop platforms.
+
+Now, the inherited `ScrollBehavior` will apply a `Scrollbar` automatically
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:17]: will apply -> applies

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  child: ListView.builder(
+    controller: controller,
+    itemBuilder: (BuildContext context, int index) {
+      return Text('Item $index');
+    }
+  )
+);
+```
+
+The `ScrollBehavior` will now apply the `Scrollbar` automatically
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:55]: will now apply -> now applies

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+when executing on desktop, and will handle providing the `ScrollController`
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:56]: will handle -> handles

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+- Extend `ScrollBehavior`, `MaterialScrollBehavior`, or `CupertinoScrollBehavior`
+to modify the default behavior.
+  
+    - With your own `ScrollBehavior`, you can apply it app-wide by setting
+      `MaterialApp.scrollBehavior`, or `CupertinoApp.scrollBehavior`.
+    - Or, if you wish to only apply it to specific widgets, add a
+      `ScrollConfiguration` above the widget in question with your
+      custom `ScrollBehavior`
+      
+Your scrollable widgets will then inherit this and reflect this behavior instead.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:100]: will then -> then

**Review by sfshaza2 — COMMENTED** [review]
Some minor tweaks, but overall looks ok.

**Comment by Piinks** [comment]
Thanks very much @sfshaza2! Updated with feedback. :)

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+`ScrollBehavior`s now automatically apply `Scrollbar`s to
+scrolling widgets on desktop platforms - Mac, Windows and Linux.
+
+## Context
+
+Prior to this change, `Scrollbar`s were applied to scrolling widgets
+manually by the developer across all platforms. This did not match
+developer expectations when executing Flutter applications on desktop platforms.
+
+Now, the inherited `ScrollBehavior` apply a `Scrollbar` automatically
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:17]:
```suggestion
Now, the inherited `ScrollBehavior` applies a `Scrollbar` automatically
```

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+
+This change did not cause any test failures, crashes, or error messages
+in the course of development, but it may result in two `Scrollbar`s
+being rendered in your application if you are manually adding `Scrollbar`s
+on desktop platforms.
+
+If you are seeing this in your application, there are several ways to
+control and configure this feature.
+
+- Remove manually applied `Scrollbar`s in your application when running on desktop.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:89]:
```suggestion
- Remove the manually applied `Scrollbar`s in your application when running on desktop.
```

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+- Extend `ScrollBehavior`, `MaterialScrollBehavior`, or `CupertinoScrollBehavior`
+to modify the default behavior.
+  
+    - With your own `ScrollBehavior`, you can apply it app-wide by setting
+      `MaterialApp.scrollBehavior`, or `CupertinoApp.scrollBehavior`.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:95]:
```suggestion
      `MaterialApp.scrollBehavior` or `CupertinoApp.scrollBehavior`.
```

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+    - Or, if you wish to only apply it to specific widgets, add a
+      `ScrollConfiguration` above the widget in question with your
+      custom `ScrollBehavior`
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:98]:
```suggestion
      custom `ScrollBehavior`.
```

File: src/docs/release/breaking-changes/default-desktop-scrollbars.md [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+      
+Your scrollable widgets then inherit this and reflect this behavior instead.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/default-desktop-scrollbars.md:100]:
```suggestion
Your scrollable widgets then inherits this and reflects this behavior.
```

**Review by sfshaza2 — APPROVED** [review]
LGTM! (Once I apply the minor suggestions.)

**Pull request merged**